concordium-sdk-examples/src/main/java/com/concordium/sdk/examples/contractexample/cis2nft/Cis2Nft.java

```diff
@@ -0,0 +1,162 @@
+                break;
+            case OPERATOR_OF:
+                SchemaParameter operatorOfParams = Cis2NftParameters.generateOperatorOfParams();
+                handleMethod(client, nonce, operatorOfParams);
+                break;
+            case BALANCE_OF:
+                SchemaParameter balanceOfParams = Cis2NftParameters.generateBalanceOfParams();
+                handleMethod(client, nonce, balanceOfParams);
+                break;
+            case TOKEN_METADATA:
+                SchemaParameter tokenMetadataParams = Cis2NftParameters.generateTokenMetadataParams();
+                handleMethod(client, nonce, tokenMetadataParams);
+                break;
+            case SUPPORTS:
+                SchemaParameter supportsParams = Cis2NftParameters.generateSupportsParameter();
+                handleMethod(client, nonce, supportsParams);
+                break;
+            case SET_IMPLEMENTORS:
+                SchemaParameter setImplementorsParams = Cis2NftParameters.generateSetImplementorsParams();
+                handleMethod(client, nonce, setImplementorsParams);
+                break;
+        }
+        return 0;
+    }
+
+    private void handleInit(ClientV2 client, Nonce nonce) {
+        InitName initName = InitName.from("init_cis2_nft");
+        InitContractPayload payload = InitContractPayload.from(CCDAmount.fromMicro(0), MODULE_REF, initName, Parameter.EMPTY);
+        InitContractTransaction initContractTransaction = TransactionFactory.newInitContract()
+                .sender(AccountAddress.from(SENDER_ADDRESS))
+                .payload(payload)
+                .expiry(EXPIRY)
+                .nonce(AccountNonce.from(nonce))
+                .signer(SIGNER)
+                .maxEnergyCost(UInt64.from(10000))
+                .build();
+        Hash txHash = client.sendTransaction(initContractTransaction);
+        System.out.println("Submitted transaction for " + this.methodName  + " with hash: " + txHash);
+        sleep();
+        Hash blockHash = client.getBlockItemStatus(txHash).getFinalizedBlockItem().get().getBlockHash();
+        System.out.println("Transaction finalized in block with hash: " + blockHash);
+    }
+
+    private void handleMethod(ClientV2 client, Nonce nonce, SchemaParameter parameter) {
+        UpdateContract payload = UpdateContract.from(CONTRACT_ADDRESS, parameter);
+        UpdateContractTransaction transaction = TransactionFactory.newUpdateContract()
+                .sender(AccountAddress.from(SENDER_ADDRESS))
+                .payload(payload)
+                .expiry(EXPIRY)
+                .nonce(AccountNonce.from(nonce))
+                .signer(SIGNER)
+                .maxEnergyCost(UInt64.from(10000))
+                .build();
+        Hash txHash = client.sendTransaction(transaction);
+        System.out.println("Submitted transaction for " + this.methodName  + " with hash: " + txHash);
+        sleep();
```

Comment:
This `sleep` is a bit hacky. 

Instead what you could do, is to query the transaction status (and block) until it's finalized. The function should also take a timeout `Duration` such that we don't end up in a situation where it blocks forever.

This could actually be a neat little helper function you could add to the `ClientV2`. We have something similar for the rust sdk.